### PR TITLE
Add workaround for JPMS edge case

### PR DESCRIPTION
**Pull request by matozoid**
"requires transitive;" now requires a module called transitive. Which will never exist. But it needs exceptions all over the parser anyway. Thanks JPMS.

See #988

### CODE REVIEW

**Comment by matozoid** [comment]
Seriously, why create such an archaic grammar for something as straightforward as this?

File: javaparser-core/src/main/java/com/github/javaparser/JavaToken.java [hunk 1]
```diff
@@ -110,6 +110,10 @@ public int getKind() {
         return kind;
     }
 
+    public void setKind(int kind) {
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/JavaToken.java]: Is it necessary to make the Token mutable? Could we instead instantiate a new token when we need to change the kind?

**Review by ftomassetti — APPROVED** [review]

**Comment by matozoid** [comment]
@ftomassetti I tried hard to make `kind` final again, but it turns out that the `JavaToken` is already stored in two places when I change the kind. One was fixable, the other was scary, so in the end I decided to only reduce acces to `package` for the setter.

**Comment by ftomassetti** [comment]
Makes sense

**Pull request merged**